### PR TITLE
Thoroughly document config.yml in README

### PR DESCRIPTION
**Pull request by MarkZH**
Proof-reading, corrections, and suggestions for moving this somewhere else are welcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,10 +51,138 @@ pip install -r requirements.txt
...
+- In `config.yml`, enter the binary name as the `engine: name` field (In Windows you may need to type a name with ".exe", like "lczero.exe")
 - Leave the `weights` field empty or see LeelaChessZero section for Neural Nets
 
+### Engine Configuration
+Besides the above, there are many possible options for configuring the engine for use with lichess-bot.
+
+- `protocol`: Specify which protocol your engine uses. Choices are
+    1. `"uci"` for the [Universal Chess Interface](http://wbec-ridderkerk.nl/html/UCIProtocol.html)
+    2. `"xboard"` for the XBoard/WinBoard/[Chess Engine Communication Protocol](https://www.gnu.org/software/xboard/engine-intf.html)
+    3. `"homemade"` if you want to write your own engine in Python within lichess-bot. See **Creating a custom bot** below.
```
Comment (TheYoBots) [inline README.md:64]: Could you link the Creating a custom bot part in markdown syntax? Like, `[**Creating a custom bot**](#creating-a-custom-bot)`

File: README.md [hunk 2]
```diff
@@ -51,10 +51,138 @@ pip install -r requirements.txt
...
+- `polyglot`: Tell lichess-bot whether your bot should use an opening book. Multiple books can be specified for each chess variant.
+    - `enabled`: Whether to use the book at all.
+    - `book`: A nested list of books. The next indented line should list a chess variant (`standard`, `3check`, `horde`, etc.) followed on succeeding indented lines with paths to the book files. See `config.yml.default` for examples.
+- `engine_options`: Command line options to pass to the engine on startup. For example, the `config.yml.default` has the configuration
+```
```
Comment (TheYoBots) [inline README.md:69]: Over here as well as in lines 105, 113, 128, 137, 164, 172 and 181 of your branch could your change it to something like, “```yml”

**Review by TheYoBots — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -51,10 +51,138 @@ pip install -r requirements.txt
...
+
+- `abort_time`: How many seconds to wait before aborting a game due to opponent inaction. This only applies during the first six moves of the game.
+- `fake_think_time`: Artificially slow down the engine to simulate a person thinking about a move. The amount of thinking time decreases as the game goes on.
+- `rate_limiting_delay`: For extremely fast games, the lichess.org servers may respond with an error if too many moves are played to quickly. This option avoids this problem by pausing for a specified number of milliseconds after submitting a move before making the next move.
+- `move_overhead`: To prevent losing on time due to network lag, substract this many milliseconds from the time to think on each move.
+
+- `correspondence` These options control how the engine behaves during correspondence games.
+  - `move_time`: How many seconds to think for each move.
+  - `checkin_period`: How often (in seconds) to reconnect to games to check for new moves after disconnecting.
+  - `disconnect_time`: How many seconds to wait after the bot makes a move for an opponent to make a move. If no move is made during the wait, disconnect from the gmae.
```
Comment (nkoppel) [inline README.md:151]: "gmae" at the end of this line is a typo

**Comment by AttackingOrDefending** [comment]
Will you also add documentation for the sections added to config.yml in #382?

**Comment by MarkZH** [comment]
@AttackingOrDefending I could, but I'm not familiar with how those options work. If you could write up what these options mean, I'll incorporate them as another commit to this PR. I'll handle formatting, too.

**Comment by AttackingOrDefending** [comment]
Yes, they do. querypv always returns the same move, which is the best one (you can verify it with queryall), query best returns more than one move and query returns many moves (not sure if they are all).
You can verify this with:
https://www.chessdb.cn/cdb.php?action=querypv&board=rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR%20w%20KQkq%20-%200%201&json=1
https://www.chessdb.cn/cdb.php?action=querybest&board=rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR%20w%20KQkq%20-%200%201&json=1
https://www.chessdb.cn/cdb.php?action=query&board=rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR%20w%20KQkq%20-%200%201&json=1
https://www.chessdb.cn/cdb.php?action=queryall&board=rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR%20w%20KQkq%20-%200%201&json=1
I think you should also add chessdb as a source for egtb in line 72.

**Comment by MarkZH** [comment]
Next draft completed.

**Review by AttackingOrDefending — APPROVED** [review]

**Pull request merged**